### ISSUE TITLE
Support Visual Studio Development

### ISSUE THREAD

**Issue by odalet** [AvaloniaUtils/ShowMeTheXaml.Avalonia#4]
Hi,

I've been trying to build [Material.Avalonia](https://github.com/AvaloniaCommunity/Material.Avalonia) from Visual Studio 2022 and it fails because of the source generator. So I tried to build the generator itself and it also fails; more accurately the sample projects fail because the source generator is simply... not running.

> **TLDR**: target `netstandard2.0` in the source generator project in order for it to be compatible with Visual Studio.

I've been digging a bit and found that anything besides `netstandard2.0` is unsupported by Microsoft for source generators or analyzers: <https://github.com/dotnet/roslyn/issues/45162#issuecomment-645052356>... and judging by the discussion in this issue and similar ones, it' unlikely this constraint will be relaxed any time soon. What's worse is that building with `dotnet build` or Rider works just fine.

Here are the error and the warning Visual Studio spits when trying to build:

```
error CS1061: 'AppBuilder' does not contain a definition for 'UseXamlDisplay' and no 
accessible extension method 'UseXamlDisplay' accepting a first argument of type 'AppBuilder' 
could be found (are you missing a using directive or an assembly reference?)

warning CS8032: An instance of analyzer ShowMeTheXaml.Avalonia.ShowMeTheXamlGenerator 
cannot be created from D:\repositories\ShowMeTheXaml.Avalonia\ShowMeTheXaml.Avalonia.Generator\bin\Debug\netstandard2.1\ShowMeTheXaml.Avalonia.Generator.dll : 
Could not load file or assembly 'netstandard, Version=2.1.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. 
Le fichier spécifié est introuvable..
```

By looking at your source code, I see two ways (none of which is optimal) for the generator to target `netstandard2.0`:

* Target `netstandard2.0` instead of `netstandard2.1` in `ShowMeTheXaml.Avalonia.Generator.csproj` then
* Either remove the dependency to `ShowMeTheXaml.Avalonia.csproj` from `ShowMeTheXaml.Avalonia.Generator.csproj`: this is the easiest way, but will prevent users from automaticallty taking a dependence on `ShowMeTheXaml.Avalonia` when they reference the generator (it might still be possible if the nuget package is handcrafted but this is a hassle...) and thus, they'll have to explicitely reference both packages.
* Or refrain from using `RuntimeFeature` in `XamlDisplay.xaml.cs` and target `netstandard2.0` in `ShowMeTheXaml.Avalonia.csproj` as well: the downside here is that `IsEditable` won't be initialized to a suitable value by default (I don't know how important this is)

Thanks for reading me

**Comment by SKProCH** [comment]
Ehh, again sufferings from VS. I'll think what i can do.

**Comment by SKProCH** [comment]
Should be fixed in 2.5.1

**Comment by odalet** [comment]
A pity that Microsoft's own flagship IDE is lagging behind Rider and even dotnet build... Anyway, thanks for this!
